helm/cluster/tests/components/test_containerd_config.go

```diff
@@ -0,0 +1,121 @@
+package components
```

Comment:
Maybe we could improve the code a bit by using testCases, because it's easier to read

```
package components

import (
    "bytes"
    "encoding/base64"
    "fmt"
    "os"
    "os/exec"
    "path/filepath"
    "runtime"
    "strings"

    . "github.com/onsi/ginkgo/v2"
    . "github.com/onsi/gomega"
)

// Embed test files
//go:embed testfiles/containerd_expected_default.toml
var expectedContainerdDefaultConfig string
//go:embed testfiles/containerd_expected_zot_mc.toml
var expectedContainerdZotMcConfig string
//go:embed testfiles/containerd_expected_zot_local.toml
var expectedContainerdZotLocalConfig string

var _ = Describe("containerd config", func() {
    type testCase struct {
        description    string
        helmValuesFile string
        expectedConfig string
    }

    // Define test cases
    testCases := []testCase{
        {
            description:    "when only required and default Helm values are used",
            helmValuesFile: "test-required-values.yaml",
            expectedConfig: expectedContainerdDefaultConfig,
        },
        {
            description:    "when MC zot is enabled",
            helmValuesFile: "test-zot-mc-values.yaml",
            expectedConfig: expectedContainerdZotMcConfig,
        },
        {
            description:    "when local zot is enabled",
            helmValuesFile: "test-zot-local-values.yaml",
            expectedConfig: expectedContainerdZotLocalConfig,
        },
    }

    for _, tc := range testCases {
        Context(tc.description, func() {
            It("renders the expected containerd config", func() {
                renderedConfig := renderContainerdConfig(tc.helmValuesFile)
                Expect(renderedConfig).To(Equal(strings.TrimSpace(tc.expectedConfig)), fmt.Sprintf("expected:\n>>>%s<<<\n, got:\n>>>%s<<<", tc.expectedConfig, renderedConfig))
            })
        })
    }
})

// render logic here similar to the JustBeforeEach block 
func renderContainerdConfig(helmValuesFile string) string {
...
}
```

Reply:
Basically with Ginko tests are written in "Ginko style" with ginko and gomega concepts, while what we have been doing with test cases approach in some projects is more like a plain Golang approach to table specs, without following any test framework's design.